### PR TITLE
Ticket 12

### PR DESCRIPTION
**Pull request by OwenPortfolio**
Add error handling for non-existant paths, and topic or article Ids.

### CODE REVIEW

**Review by emilyb93 — COMMENTED** [review]
You have a bug where if you go to 'articles/topics/notatopic' , you get the error 404, but you cannot actually remove the error by clicking the topic links.

**Review by emilyb93 — COMMENTED** [review]
You also have a bug where going to something '/articles/fsfasf' it doesnt show an error, you get a react issue

File: src/components/SingleTopic.jsx [hunk 1]
```diff
@@ -9,16 +9,25 @@ const SingleTopic = () => {
...
     const [sortOrder, setSortOrder] = useState('desc');
+    const [error, setError] = useState();
 
     useEffect(() => {
         getArticlesByTopic(topic, sort, sortOrder).then((articlesByTopicFilter) => {
-            setTopicFilter(articlesByTopicFilter.articles);
-            setLoading(false);
+            if(articlesByTopicFilter.articles){
+                setTopicFilter(articlesByTopicFilter.articles);
+                setLoading(false);
```
Comment (emilyb93) [inline src/components/SingleTopic.jsx:18]: You need to setError to false here as well or you can get caught in an error page

File: src/components/Article.jsx [hunk 2]
```diff
@@ -38,6 +43,8 @@ const Article = ({user}) => {
 
     if(loading){
         return (<h1>LOADING</h1>)
+    } else if(error === 404){
```
Comment (emilyb93) [inline src/components/Article.jsx:46]: The error might not be a 404 if we make a bad request with requesting an id of a string, we get a 400 which is causing this error component to not work correctly. A better way would be to just say if(error)

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Some error bugs

**Comment by OwenPortfolio** [comment]
All fixed (I hope)

**Comment by dappernerddesigns** [comment]
All bugs smooshed. Merge away :)

**Comment by OwenPortfolio** [comment]
Glad to see you're not slacking on your last day. ;) Thanks

**Pull request merged**